File: Best_Time_to_Buy_and_Sell_Stock/submission3.py
class Solution(object):
    def maxProfit(self, prices):
        
        ans = 0
        largest = prices[-1]
        n = len(prices)

        for i in range(n-2, -1, -1):
            ans = max(ans, largest-prices[i])
            largest = max(largest, prices[i])
        
        return ans

File: Reverse_Words_in_a_String/submission3.cpp
class Solution {
public:
    string reverseWords(string s) {
        
        int size = s.size();
        int i=size-1;
        string ans = "";

        while (i>=0){

            while (i>=0 && s[i]==' '){
                i--;
            }

            int j=i;
            while (j>=0 && s[j]!=' '){
                j--;
            }

            for (int k=j+1; k<=i; k++){
                ans.push_back(s[k]);
            }

            i=j;
            while (i>=0 && s[i]==' '){
                i--;
            }

            if (i!=-1){
                ans.push_back(' ');
            }
        }

        return ans;

    }
};